### ISSUE TITLE
1.19.6: test_checkbox and/or test_combobox lead to SIGABRT ("corrupted double-linked list")

### ISSUE THREAD

**Issue by dvzrv** [pymupdf/PyMuPDF#1663]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
When trying to package 1.19.6 for Arch Linux I attempted to run the tests again.
It appears that [test_checkbox()](https://github.com/pymupdf/PyMuPDF/blob/caf04c277b73941d6552ed46ec55d1abc04e9ce8/tests/test_widgets.py#L45-L60) and [test_combobox()](https://github.com/pymupdf/PyMuPDF/blob/caf04c277b73941d6552ed46ec55d1abc04e9ce8/tests/test_widgets.py#L92-L125) both lead to SIGABRT of python/pytest after the tests are finished.
The following is printed, after the printout of the pytest result overview (example where test_checkbox() leads to the crash):

```
corrupted double-linked list
/startdir/PKGBUILD: line 24:  1301 Aborted                 (core dumped) pytest -vv -k "not test_pageids and not test_textbox3 and not test_combobox"
```

## To Reproduce (mandatory)

```
local _site_packages=$(python -c "import site; print(site.getsitepackages()[0])")
local _test_dir="test_dir"
python -m installer --destdir="$_test_dir" dist/*.whl
export PYTHONPATH="$_test_dir/$_site_packages:$PYTHONPATH"
pytest -vv -k "not test_pageids and not test_textbox3 and not test_combobox"
```

## Expected behavior (optional)
Tests pass and pytest/ python does not SIGABRT.

## Screenshots (optional)
n/a

## Your configuration (mandatory)
 - Operating system, potentially version and bitness: Arch Linux (x86_64)
 - Python version, bitness: 3.10.4 (x86_64)
 - PyMuPDF version, installation method (**wheel** or **generated** from source): 1.19.6 (attempting to build from pypi sdist)

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

## Additional context (optional)
I am able to ignore the two test functions (there are two others which I have issues with, but for other reasons and they do not lead to SIGABRT).

**Comment by julian-smith-artifex-com** [comment]
Would you be able to test with PyMuPDF-1.20.0 (released a couple of days ago)?

This has a fix for a potential SEGV that could be related to what you're seeing here.

**Comment by dvzrv** [comment]
Yes, this is fixed with pymupdf 1.20.0 and building against mupdf/libmupdf 1.20.0. :tada:
